Fix class and section status inconsistencies

willgearty opened the pull request:
This fixes a few separate issues, some of which were reported:

1. Previously we had changed `isAccepted` for classes to also return `True` if the class was "accepted but hidden. I've copied this behavior over to sections.
2. We weren't including rejected classes on the teacher registration and userview pages.
3. We weren't distinguishing between "Cancelled" and "Rejected" status on the teacher registration and userview pages.
4. We weren't identifying "accepted but hidden" classes or sections as "hidden" on the teacher registration and userview pages.

It's certainly possible there are other places we repeat numbers 2 through 3, so please let me know if anything comes to mind during review!

Fixes https://github.com/learning-unlimited/ESP-Website/issues/2992 and fixes https://github.com/learning-unlimited/ESP-Website/issues/3003.

kkbrum commented:
1. Lower on the manage page in the dropdowns for the class options, it says "accepted but hidden" but at the top of the manage page, it still just says accepted. could we change that to specify the hiddenness?

![image](https://user-images.githubusercontent.com/25392075/89596459-356dcf80-d825-11ea-8904-ae1a076c467d.png

2. When i changed the status of the class on the manage page in the class options box from accepted (or rejected) to accepted but hidden, it did not change the status of the section as appearing in the "information for section 1" box on the manage page. This seems to contradict what I expected and the note above the class options box that says "Please use the form below to alter blah blah. Note that changes made to the class will propagate to each of the sections, but you can later change each section independently."

willgearty commented:
OK, I can't make the page override the section statuses without breaking the ability to have custom section statuses at all (without a much bigger change to how we handle the forms on the manage class page which is definitely outside the scope of this PR). So instead I modified the help text to reflect the actual behavior. If you definitely want the other behavior, feel free to make a separate issue. I also fixed the status at the top of the page as requested.
